src/services/goals.js

```diff
@@ -1267,6 +1267,13 @@ async function removeObjectives(objectivesToRemove, reportId) {
     return Promise.resolve();
   }
 
+  // cleanup any ObjectiveFiles that are no longer needed
+  await ObjectiveFile.destroy({
+    where: {
+      objectiveId: objectivesToDefinitelyDestroy.map((o) => o.id),
+    },
+  });
+
```

Comment:
Do we need to add resource and topic deletes aswell?

Reply:
Looks like they were set to "onDelete: cascade" where this model wasn't

Reply:
That's a better fix